libs/s25client/s25client.cpp

```diff
@@ -278,45 +280,85 @@ void SetAppSymbol()
 #endif // _WIN32
 }
 
-bool InitDirectories()
+bool MigrateFilesAndDirectories()
 {
-    // Note: Do not use logger yet. Filepath may not exist
-    const std::string curPath = bfs::current_path().string();
-    LOG.write("Starting in %s\n", LogTarget::Stdout) % curPath;
-
-    // diverse dirs anlegen
-    std::array<unsigned, 9> dirs = {{94, 41, 47, 48, 51, 85, 98, 99, 100}}; // settingsdir muss zuerst angelegt werden (94)
-
-    std::string oldSettingsDir;
-
+    struct MigrationEntry
+    {
+        std::string oldName, newName;
+        bool isFolder;
+    };
+    // Mapping from old files or directories to new ones
+    // Handled in order so multiple renamings are possible
+    // Empty newName = Delete
+    const std::vector<MigrationEntry> migrations = {
 #ifdef _WIN32
-    oldSettingsDir = RTTRCONFIG.ExpandPath("~/Siedler II.5 RttR");
+      {"~/Siedler II.5 RttR", s25::folders::config, true},
 #elif defined(__APPLE__)
-    oldSettingsDir = RTTRCONFIG.ExpandPath("~/.s25rttr");
+      {"~/.s25rttr", s25::folders::config, true},
 #endif
-    if(!oldSettingsDir.empty() && bfs::is_directory(oldSettingsDir))
+      {std::string(s25::folders::lstsUser).append("/SOUND.LST"), "", false},
```

Comment:
we should add the old SDL drivers here too

Reply:
Done although it is a bit dirty due to os-specific extensions